server: TestMakeIdleMonitor_WithBriefConnectionDuringWarmup failed

[(server).TestMakeIdleMonitor_WithBriefConnectionDuringWarmup failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2671714&tab=buildLog) on [master@ea1ef0b9af4345ffb90a03dd21b59c099cd1f41d](https://github.com/cockroachdb/cockroach/commits/ea1ef0b9af4345ffb90a03dd21b59c099cd1f41d):

```
=== RUN   TestMakeIdleMonitor_WithBriefConnectionDuringWarmup
    idle_monitor_test.go:176: 
        	Error Trace:	idle_monitor_test.go:176
        	Error:      	Should be true
        	Test:       	TestMakeIdleMonitor_WithBriefConnectionDuringWarmup
--- FAIL: TestMakeIdleMonitor_WithBriefConnectionDuringWarmup (0.32s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestMakeIdleMonitor_WithBriefConnectionDuringWarmup PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestMakeIdleMonitor_WithBriefConnectionDuringWarmup.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(server).TestMakeIdleMonitor_WithBriefConnectionDuringWarmup failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2673033&tab=buildLog) on [master@0bf4147af00cf298986cd5944fde51bc8ec26e4e](https://github.com/cockroachdb/cockroach/commits/0bf4147af00cf298986cd5944fde51bc8ec26e4e):

```
=== RUN   TestMakeIdleMonitor_WithBriefConnectionDuringWarmup
    idle_monitor_test.go:169: 
        	Error Trace:	idle_monitor_test.go:169
        	Error:      	Should be false
        	Test:       	TestMakeIdleMonitor_WithBriefConnectionDuringWarmup
--- FAIL: TestMakeIdleMonitor_WithBriefConnectionDuringWarmup (0.32s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestMakeIdleMonitor_WithBriefConnectionDuringWarmup PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestMakeIdleMonitor_WithBriefConnectionDuringWarmup.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(server).TestMakeIdleMonitor failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2674357&tab=buildLog) on [master@ba1a1440714b8c514f8a994b0a82c426c3ee6e99](https://github.com/cockroachdb/cockroach/commits/ba1a1440714b8c514f8a994b0a82c426c3ee6e99):

```
=== RUN   TestMakeIdleMonitor
    idle_monitor_test.go:48: 
        	Error Trace:	idle_monitor_test.go:48
        	Error:      	Should be false
        	Test:       	TestMakeIdleMonitor
--- FAIL: TestMakeIdleMonitor (0.34s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestMakeIdleMonitor PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestMakeIdleMonitor.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(server).TestMakeIdleMonitor failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2673888&tab=buildLog) on [master@9969862335cf501c1aa67a4aaf018ad5dc8e85e8](https://github.com/cockroachdb/cockroach/commits/9969862335cf501c1aa67a4aaf018ad5dc8e85e8):

```
=== RUN   TestMakeIdleMonitor
    idle_monitor_test.go:55: 
        	Error Trace:	idle_monitor_test.go:55
        	Error:      	Should be true
        	Test:       	TestMakeIdleMonitor
--- FAIL: TestMakeIdleMonitor (0.32s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestMakeIdleMonitor PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestMakeIdleMonitor.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/server/idle_monitor_test.go
+++ b/pkg/server/idle_monitor_test.go
@@ -15,21 +15,26 @@ import (
 	"testing"
 	"time"
 
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/stretchr/testify/require"
 )
 
+var (
+	warmupDuration    = 600 * time.Millisecond
+	countdownDuration = 1200 * time.Millisecond
+	delta             = 150 * time.Millisecond
+)
+
 func TestMakeIdleMonitor(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	skip.UnderShort(t)
 
 	ctx := context.Background()
 
 	start := timeutil.Now()
-	warmupDuration := 300 * time.Millisecond
-	countdownDuration := 600 * time.Millisecond
-	delta := 5 * time.Millisecond
 
 	var handlerCalled syncutil.AtomicBool
 	monitor := MakeIdleMonitor(ctx, warmupDuration, func() {
@@ -77,13 +82,11 @@ func TestMakeIdleMonitor(t *testing.T) {
 
 func TestMakeIdleMonitor_WithConnectionDuringWarmup(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	skip.UnderShort(t)
 
 	ctx := context.Background()
 
 	start := timeutil.Now()
-	warmupDuration := 300 * time.Millisecond
-	countdownDuration := 600 * time.Millisecond
-	delta := 5 * time.Millisecond
 
 	var handlerCalled syncutil.AtomicBool
 	monitor := MakeIdleMonitor(ctx, warmupDuration, func() {
@@ -136,13 +139,11 @@ func TestMakeIdleMonitor_WithConnectionDuringWarmup(t *testing.T) {
 
 func TestMakeIdleMonitor_WithBriefConnectionDuringWarmup(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	skip.UnderShort(t)
 
 	ctx := context.Background()
 
 	start := timeutil.Now()
-	warmupDuration := 300 * time.Millisecond
-	countdownDuration := 600 * time.Millisecond
-	delta := 5 * time.Millisecond
 
 	var handlerCalled syncutil.AtomicBool
 	monitor := MakeIdleMonitor(ctx, warmupDuration, func() {
@@ -198,13 +199,11 @@ func TestMakeIdleMonitor_WithBriefConnectionDuringWarmup(t *testing.T) {
 
 func TestMakeIdleMonitor_WithBriefConnectionDuringCountdown(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	skip.UnderShort(t)
 
 	ctx := context.Background()
 
 	start := timeutil.Now()
-	warmupDuration := 300 * time.Millisecond
-	countdownDuration := 600 * time.Millisecond
-	delta := 5 * time.Millisecond
 
 	var handlerCalled syncutil.AtomicBool
 	monitor := MakeIdleMonitor(ctx, warmupDuration, func() {

